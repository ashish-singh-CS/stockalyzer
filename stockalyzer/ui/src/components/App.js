import React, { Component, Fragment } from "react";
import { AppBar } from "./app_bar"

export default class extends Component {
    render() {
      return (
        <Fragment>
          <AppBar /> 
        </Fragment>
      );
    }
  }