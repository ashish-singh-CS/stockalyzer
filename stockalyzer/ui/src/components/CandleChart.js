import React from "react";
import ReactApexChart from "react-apexcharts";

const options = {
  chart: {
    type: 'candlestick',
    height: 350
  },
  title: {
    text: 'OHLC',
    align: 'left'
  },
  xaxis: {
    type: 'datetime'
  },
  yaxis: {
    tooltip: {
      enabled: true
    }
  }
}

export default function (props) {
  
  return (
<div id="chart">
<ReactApexChart options={options} series={props.series} type="candlestick" height={"350"} />
</div>
  );
}
