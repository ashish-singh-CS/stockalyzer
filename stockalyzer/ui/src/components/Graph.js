import React, { Component } from 'react';
import Plot from 'react-plotly.js';

export default class extends Component {
  render() {
    return (
      <Plot
        data={[
          {
            x: [1, 2, 3],
            y: [2, 6, 3],
            type: 'scatter',
            mode: 'lines+markers',
            marker: {color: 'red'},
          },
          {type: 'bar', x: [1, 2, 3], y: [2, 5, 3]},
        ]}
        layout={ {width: 640, height: 420, title: 'A Fancy Plot'} }
      />
    );
  }
}