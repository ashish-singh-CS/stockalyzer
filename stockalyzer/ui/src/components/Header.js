import React from "react";
import { Typography } from "@material-ui/core";

/**
 * Header that displays the company name and symbol 
 */

export default function Header(props) {
    return(
        <Typography variant = "h5"> 
            <b>{props.company.name} {"(" + props.company.symbol + ")"}</b>
        </Typography>
    )
}