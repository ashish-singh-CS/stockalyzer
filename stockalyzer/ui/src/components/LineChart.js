import React from "react";
import ReactApexChart from "react-apexcharts";

const options = {
    chart: {
      type: 'area',
      stacked: false,
      height: 350,
      zoom: {
        type: 'x',
        enabled: true,
        autoScaleYaxis: true
      },
      toolbar: {
        autoSelected: 'zoom'
      }
    },
    dataLabels: {
      enabled: false
    },
    markers: {
      size: 0,
    },
    title: {
      text: 'Technical Indicators',
      align: 'left'
    },
    fill: {
      type: 'gradient',
      gradient: {
        shadeIntensity: 1,
        inverseColors: false,
        opacityFrom: 0.5,
        opacityTo: 0,
        stops: [0, 90, 100]
      },
    },
    yaxis: {
      labels: {
        formatter: function (val) {
          return (val / 1000000).toFixed(0);
        },
      },
      title: {
        text: 'Price'
      },
    },
    xaxis: {
      type: 'datetime',
    },
    tooltip: {
      shared: false,
      y: {
        formatter: function (val) {
          return val
        }
      }
    }
}


export default function (props) {
  
  return (
  <div id="chart">
    <ReactApexChart options={options} series={props.series} type="area" height={350} />
  </div>
  );
}