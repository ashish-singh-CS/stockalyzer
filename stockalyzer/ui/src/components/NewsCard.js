import React from 'react';
import { makeStyles, Link } from '@material-ui/core';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

/**
 * The media card that displays news information
 * Template Source: https://github.com/mui-org/material-ui/blob/master/docs/src/pages/components/cards/ImgMediaCard.js
 */

const useStyles = makeStyles({
  root: {
    maxWidth: 600,
    width: 600,
  }
});

export default function MediaCard(props) {
  const classes = useStyles();

  return (
    <div style={{paddingTop:20, paddingLeft: 20}}>
    <Card className={classes.root}>
      <CardActionArea>
        <CardContent height="200">
          <Typography gutterBottom variant="h5" component="h2">
            News
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {props.news.title}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions style={{ paddingTop: 50 }}>
        <Button size="small" color="primary" onClick={props.handleBack} disabled={props.backDisabled}>
          Back
        </Button>
        <Button size="small" color="primary" onClick={props.handleNext} disabled={props.nextDisabled}>
          Next
        </Button>
        <Button size="small" color="primary" component={Link} href={props.news.url} disabled={props.linkDisabled}>
          Learn More
        </Button>
      </CardActions>
    </Card>
    </div>
  );
}
