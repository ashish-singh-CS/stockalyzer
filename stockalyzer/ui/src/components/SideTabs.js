import React from "react";
import PropTypes from "prop-types";
import { makeStyles, Tabs, Tab, Typography, Box } from "@material-ui/core";

/**
 * The tabs on the side of the page with a list of stocks
 * Template Source: https://github.com/mui-org/material-ui/blob/master/docs/src/pages/components/tabs/VerticalTabs.js
 */

function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <Typography
      component="div"
      role="tabpanel"
      hidden={value !== index}
      id={`vertical-tabpanel-${index}`}
      aria-labelledby={`vertical-tab-${index}`}
      {...other}
    >
      {value === index && <Box p={3}>{children}</Box>}
    </Typography>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired
};

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.paper,
    display: "fixed",
    height: 800,
    width: 270
  },
  tabs: {
    borderRight: `1px solid ${theme.palette.divider}`
  }
}));

export default function VerticalTabs(props) {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
    props.tabChange(newValue);
  };

  return (
    <div className={classes.root}>
      <Tabs
        orientation="vertical"
        variant="scrollable"
        value={value}
        onChange={handleChange}
        aria-label="Vertical tabs example"
        className={classes.tabs}
      >

        {props.tickers.map( (ticker) => 
          <Tab key = {ticker.id} label = {ticker.company}> </Tab>
        )}

      </Tabs>
    </div>
  );
}