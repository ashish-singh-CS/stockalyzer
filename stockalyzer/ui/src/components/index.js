import AppBar from "./AppBar";
import SideTabs from "./SideTabs";
import Header from "./Header"
import Table from "./Table"
import TweetCounts from "./TweetCounts"
import Graph from "./Graph"

export { AppBar, SideTabs, Header, Table, TweetCounts, Graph };