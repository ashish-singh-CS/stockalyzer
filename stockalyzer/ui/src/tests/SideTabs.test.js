import React from 'react';
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect'
import {SideTabs} from "../components";

const tickers = [];

afterEach(cleanup);

it("renders", () => {
    const {asFragment} = render(<SideTabs tickers={tickers} />);
    expect(asFragment()).toMatchSnapshot(); 
});