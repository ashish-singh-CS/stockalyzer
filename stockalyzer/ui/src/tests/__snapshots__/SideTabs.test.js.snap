// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      aria-label="Vertical tabs example"
      class="MuiTabs-root makeStyles-tabs-2 MuiTabs-vertical"
    >
      <div
        class="MuiTabs-scrollable"
        style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
      />
      <div
        class="MuiTabs-scroller MuiTabs-scrollable"
        style="margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer MuiTabs-flexContainerVertical"
          role="tablist"
        />
        <span
          class="PrivateTabIndicator-root-14 PrivateTabIndicator-colorSecondary-16 MuiTabs-indicator PrivateTabIndicator-vertical-17"
          style="top: 0px; height: 0px;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
